### ISSUE TITLE
Get routing information

### ISSUE THREAD

**Issue by eyip002** [uniba-swt/swtbahn-cli#48]
- ~~Update extra_config.yml with better orientation information for overlaps and signals involved~~
- ~~Add a monitor function (`get_block(block_id, orientation)`) to return an array of signal and segments in their order of occurrence according to orientation~~
- [x] Update YAML parser for interlocking table (signals) ~~and extras (signals and segments)~~
- [x] Implement `bool is_type_segment(const char *id)`
- [x] Implement `bool is_type_signal(const char *id)`
- [x] ~~Extend the monitor with methods to query whether an id is a segment or signal~~

**Comment by BLuedtke** [comment]
I implemented a 'simulated' route information handling of a client in some of my MBT models. So, just as an example (and for the record) of what I would find "useful" to be able to get:
When I successfully request a route, I receive the ID of that route. Using this ID, I'd like to get more information about the route. One 'form' of such information could be the route in what I call "sequential block notation". This is effectively just a transformation of the commented line that appears in the interlocking table after the line "path:" for each route.

So, for the route with ID 1 on swtbahn-standard (see [interlocking table](https://github.com/uniba-swt/swtbahn-cli/blob/master/configurations/swtbahn-standard/interlocking_table.yml)), this could be a string like such:       
`"point6.reverse,block4,point8.reverse,point2.reverse,block2,point3.reverse,point4.reverse,point5.reverse,crossing1.line2,point12.reverse1,point11.normal,point9.reverse,point10.reverse,block7,point12.reverse2,platform2"`

The assumption is that the order of this string corresponds to the order in which this route would be driven. Whitespace after each comma wouldn't be an issue.

After retrieving that info, for my own use, I prepend the start signal and the start signals "overlap segment" to that string. Probably don't need an extra endpoint for that. Use case is a check for possible obstacles, where I need to check for a possible obstacle on the overlap segment after the start signal too.

**Comment by eyip002** [comment]
@BLuedtke What do you think about the following, where `path` now includes intermediate signals, and `signals` includes all signals of the route:

```yaml
# Interlocking table
interlocking-table:
  - id: 0 #route0
    source: signal8
    destination: signal19
    orientation: anticlockwise
    path:
      # point6.reverse -> block4 -> point8.reverse -> point2.reverse -> block2 -> point3.reverse -> point4.reverse -> point5.normal -> point9.normal -> point10.reverse -> block7 -> point12.reverse2 -> platform2
      - id: seg14
        type: segment
      - id: seg15
        type: segment
      - id: seg16
        type: segment
      - id: seg17
        type: segment
      - id: signal12
        type: signal
      - id: seg18
        type: segment
      - id: seg6
        type: segment
      - id: seg7
        type: segment
      - id: seg8
        type: segment
      - id: seg9
        type: segment
      - id: signal6
        type: signal
      - id: seg10
        type: segment
      - id: seg23
        type: segment
      - id: seg24
        type: segment
      - id: seg25
        type: segment
      - id: seg26
        type: segment
      - id: seg30b
        type: segment
      - id: seg31
        type: segment
      - id: seg32b
        type: segment
      - id: signal15
        type: signal
      - id: seg34
        type: segment
      - id: seg39
        type: segment
      - id: seg40
        type: segment
      - id: seg41
        type: segment
    sections:
      - id: block4
      - id: block2
      - id: block7
      - id: platform2
    length: 595.60cm
    signals:
      - id: signal8
      - id: signal12
      - id: signal6
      - id: signal15
      - id: signal19
    points:
      - id: point6
        position: reverse
      - id: point8
        position: reverse
      - id: point2
        position: reverse
      - id: point3
        position: reverse
      - id: point4
        position: reverse
      - id: point5
        position: normal
      - id: point9
        position: normal
      - id: point10
        position: reverse
      - id: point12
        position: reverse
```

**Comment by BLuedtke** [comment]
> 

@eyip002  That'd work great. Path and Points combined would contain all information I need to control interlocking and driving in the models, which is very convenient.

Do we want to think/consider Flank protection at this point? I think we can defer that to a future release tbh.

**Comment by eyip002** [comment]
No need to extend the monitor with methods to query whether an id is a segment or signal, because a list of signals and segments can already be retrieved with the monitor.

**Comment by eyip002** [comment]
Implemented in commit 814c1e7
